__TESTS__/unit/config/cloudinaryConfig.test.ts

```diff
@@ -152,4 +155,167 @@ describe('Tests for CloudinaryConfiguration', () => {
 
     expect(url).toContain('http://');
   });
-});
+  it('should allow overriding cloudName in config', () => {
+    const conf = new CloudinaryConfig({
+      cloud: {
+        cloudName: 'foo'
+      }
+    });
+
+    conf.cloud.cloudName = 'demo';
+
+    expect(conf.cloud).toEqual({
+      cloudName: 'demo'
+    });
+  });
+
+  // These tests should be "translated" to js base code when these missing features are added:
+  it.skip('should use resource_type from options', () => {
+    test_cloudinary_url('test', {
+      resource_type: 'raw'
+    }, 'https://res.cloudinary.com/test123/raw/upload/test', {});
+  });
+
+  it.skip('should support external cname with cdn_subdomain on', () => {
+    test_cloudinary_url('test', {
+      cname: 'hello.com',
+      cdn_subdomain: true
+    }, 'https://a2.hello.com/test123/image/upload/test', {});
+  });
+  it.skip('should allow to shorted image/upload urls', () => {
+    test_cloudinary_url('test', {
+      shorten: true
```

Comment:
Can you please implement the full test?

You can set the shorten property, but it does not affect teh URL at this point. (meaning the config is settable, but the URL is not modified because of it)